x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportPutTrainedModelAction.java

```diff
@@ -230,7 +230,8 @@ protected void masterOperation(
         if (TrainedModelAssignmentMetadata.fromState(state).hasDeployment(trainedModelConfig.getModelId())) {
             finalResponseListener.onFailure(
                 ExceptionsHelper.badRequestException(
-                    "Cannot create model [{}] the id is the same as an current model deployment",
+                    // The inference service depends on this error message to determine if the model is already downloaded
+                    "Cannot create model [{}] the model id is the same as the deployment id of a current model deployment",
```

Comment:
What do you think about moving the portion of the message that `TextEmbeddingInternalService` needs to https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/PutTrainedModelAction.java#L29

`the model id is the same as the deployment id of a current model deployment`

I think the inference API can access that ok without needing the ml plugin loaded 🤔 

And then in here we can just concatenate that with this message.